### PR TITLE
Remove likely unnecessary default_name, fixes broken 2023.9

### LINKED ISSUE #119: Home Assistant 2023.8.0: Invalid Device Info
Log:

```
2023-08-02 17:13:13.904 ERROR (MainThread) [homeassistant.components.sensor] Ignoring invalid device info: Invalid device info {'configuration_url': 'http://XXX.XXX.XX.XX:XXXX/children/xxxxx/dashboard/', 'default_name': 'Baby XXXXX ', 'identifiers': {('babybuddy', 2)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.
```

It appears that all entities are separated from the device, causing the Add Feeding and Add Change services to break if you use the device as the target. I am unsure if it affects other services.

### PR DESCRIPTION
**Pull request by gagebenne**
Closes #119.

Could add child slug (Baby Buddy API unique identifier) in case you want more unique identifiers.

### CODE REVIEW

**Comment by jcgoette** [comment]
This works ([for whatever reason??](https://developers.home-assistant.io/docs/device_registry_index/#categorizing-to-device-info)), but let's change to `name` instead of removing completely.

File: custom_components/babybuddy/switch.py [hunk 1]
```diff
@@ -148,7 +148,6 @@ def __init__(
         self._attr_icon = "mdi:timer-sand"
         self._attr_device_info = {
             "identifiers": {(DOMAIN, child[ATTR_ID])},
-            "default_name": f"{child[ATTR_FIRST_NAME]} {child[ATTR_LAST_NAME]}",
         }
```
Comment (gagebenne) [inline custom_components/babybuddy/switch.py:151]:
```suggestion
ATTR_SLUG
            "name": f"{child[ATTR_SLUG]}",
        }
```

File: custom_components/babybuddy/sensor.py [hunk 2]
```diff
@@ -203,7 +203,6 @@ def __init__(self, coordinator: BabyBuddyCoordinator, child: dict) -> None:
         self.child = child
         self._attr_device_info = {
             "configuration_url": f"{coordinator.config_entry.data[CONF_HOST]}:{coordinator.config_entry.data[CONF_PORT]}{coordinator.config_entry.data[CONF_PATH]}/children/{child[ATTR_SLUG]}/dashboard/",
-            "default_name": f"Baby {child[ATTR_FIRST_NAME]} {child[ATTR_LAST_NAME]}",
             "identifiers": {(DOMAIN, child[ATTR_ID])},
```
Comment (gagebenne) [inline custom_components/babybuddy/sensor.py:206]:
```suggestion
            "identifiers": {(DOMAIN, child[ATTR_ID])},
            "name": f"{child[ATTR_SLUG]}",
```

File: custom_components/babybuddy/switch.py [hunk 3]
```diff
@@ -148,7 +148,6 @@ def __init__(
         self._attr_icon = "mdi:timer-sand"
         self._attr_device_info = {
             "identifiers": {(DOMAIN, child[ATTR_ID])},
```
Comment (gagebenne) [inline custom_components/babybuddy/switch.py:150]:
```suggestion
            "identifiers": {(DOMAIN, child[ATTR_ID])},
            "name": f"{child[ATTR_SLUG]}",
```

File: custom_components/babybuddy/sensor.py (hunk 2 above)
Comment (jcgoette) [reply custom_components/babybuddy/sensor.py:206]: Let's keep `f"{child[ATTR_FIRST_NAME]} {child[ATTR_LAST_NAME]}"` format.

File: custom_components/babybuddy/switch.py (hunk 3 above)
Comment (jcgoette) [reply custom_components/babybuddy/switch.py:150]: Let's keep f"{child[ATTR_FIRST_NAME]} {child[ATTR_LAST_NAME]}" format.

**Comment by gagebenne** [comment]
> Ha, yea, I just noticed sensor.py had "Baby" prefix and switch.py didn't. I don't feel strongly either way and I think it will either break nothing or break things equally.

Just ran these changes, nothing seemed to break in my setup!

**Comment by jcgoette** [comment]
I meant by changing both to be consistent. Which we should do.

**Pull request merged**